### PR TITLE
fixed json type and added server properties

### PR DESCRIPTION
**Pull request by rymaju**
I changed all responses such that they use the static end() methods shamelessly stolen from the backend scaffold.

Also, when changing the default port to 80 for devops, it made local testing difficult. I just added a server properties file so that we can set the port to whatever we want, defaulting to the old 8090 for local development.

### CODE REVIEW

**Comment by rymaju** [comment]
image

**Comment by rymaju** [comment]
image

File: README.md [hunk 1]
```diff
@@ -42,7 +42,23 @@ $ CREATE DATABASE c4cneu-db;
...
+database.password = root
+jwt.secret = SUPER_SECRET_KEY
+jwt.duration = 3600000
+
+```
+
+Update the properties file in `/persist/src/main/resources/server.properties` to contain the port the server should run on
+
+```
+server.port = 8090
```
Comment (liammoyn) [inline README.md:58]: 8080?

File: api/src/main/java/com/codeforcommunity/ApiMain.java [hunk 2]
```diff
@@ -45,6 +45,7 @@ public void startHttpsServerApi() {
 
     HttpServer server = vertx.createHttpServer(serverOptions);
     Router router = apiRouter.initializeRouter(vertx);
-    server.requestHandler(router).listen(80);
+    System.out.println("Server listening on port " + serverProperties.getProperty("server.port"));
+    server.requestHandler(router).listen(Integer.parseInt(serverProperties.getProperty("server.port")));
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/ApiMain.java]: You should probably initialize an integer variable called port and then pass that to listen. Just to make the code a little bit more readable

File: api/src/main/java/com/codeforcommunity/dto/IRequest.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.codeforcommunity.dto;
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/dto/IRequest.java]: As far as I can tell this class is unused, why is it included?

File: api/src/main/java/com/codeforcommunity/dto/IResponse.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.codeforcommunity.dto;
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/dto/IResponse.java]: Same with this class, looks unnecessary

**Review by liammoyn — DISMISSED** [review]
Looks fine but please read my comments

File: README.md (hunk 1 above)
Comment (rymaju) [reply README.md:58]: 8090 was what the default port was changed to during the first production merge, and it was changed again to 80 during the second production merge. It made testing a little bit weird.

In retrospect, this port should really be a completely unused port, so 8080 or 8090 is also inappropriate.

Maybe we should do something like 5000 or 8000 instead?

File: api/src/main/java/com/codeforcommunity/ApiMain.java (hunk 2 above)
Comment (rymaju) [reply api/src/main/java/com/codeforcommunity/ApiMain.java]: Yep good idea

File: api/src/main/java/com/codeforcommunity/dto/IRequest.java (hunk 3 above)
Comment (rymaju) [reply api/src/main/java/com/codeforcommunity/dto/IRequest.java]: Yep I dont think either are used, ill get rid of both

File: README.md (hunk 1 above)
Comment (liammoyn) [reply README.md:58]: It doesn't really matter, I think just picking one and rolling with it is probably a fine approach

**Review by liammoyn — APPROVED** [review]

**Pull request merged**